### PR TITLE
Remove fakeredis version constraint

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation

Fix #3607 .
Currently CI is failing due to old fakeredis.

## Description of the changes
Remove all constraints of fakeredis.

### CODE REVIEW

**Comment by contramundum53** [comment]
@c-bata Can you review this PR? 
Since the changes are only related to CI, you can merge this PR by one accept.

File: .github/workflows/tests-integration.yml [hunk 1]
```diff
@@ -59,7 +59,6 @@ jobs:
         pip install --progress-bar off .[test]
         pip install --progress-bar off .[optional]
         pip install --progress-bar off .[integration] --extra-index-url https://download.pytorch.org/whl/cpu
-        pip install "fakeredis<2.11.1"
```
Comment (c-bata) [inline .github/workflows/tests-integration.yml:62]: I think you can remove a comment in L49.

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -55,7 +55,6 @@ jobs:
         pip install --progress-bar off .[test]
         pip install --progress-bar off .[optional]
         pip install --progress-bar off .[integration] --extra-index-url https://download.pytorch.org/whl/cpu
-        pip install "fakeredis<2.11.1"
```
Comment (c-bata) [inline .github/workflows/coverage.yml:58]: Could you remove the comment in L45?

File: .github/workflows/mac-tests.yml [hunk 3]
```diff
@@ -52,7 +52,6 @@ jobs:
         optuna --version
         pip install --progress-bar off .[test]
         pip install --progress-bar off .[optional]
-        pip install "fakeredis<2.11.1"
```
Comment (c-bata) [inline .github/workflows/mac-tests.yml:55]: ditto.

File: .github/workflows/mac-tests.yml [hunk 4]
```diff
@@ -116,7 +115,6 @@ jobs:
         pip install --progress-bar off .[test]
         pip install --progress-bar off .[optional]
         pip install --progress-bar off .[integration]
-        pip install "fakeredis<2.11.1"
```
Comment (c-bata) [inline .github/workflows/mac-tests.yml:119]: ditto.

**Review by c-bata — CHANGES_REQUESTED** [review]
Thank you for your pull request. I left some minor comments.

**Review by c-bata — APPROVED** [review]
LGTM.

**Pull request merged**